flood_adapt/object_model/hazard/event/hurricane.py

```diff
@@ -0,0 +1,108 @@
+from pathlib import Path
+from typing import List
+
+import pyproj
+from cht_cyclones.tropical_cyclone import TropicalCyclone
+from pydantic import BaseModel
+from shapely.affinity import translate
+
+import flood_adapt.dbs_controller as db
+from flood_adapt.object_model.hazard.interface.events import IEvent, IEventModel
+from flood_adapt.object_model.hazard.interface.forcing import (
+    ForcingSource,
+    ForcingType,
+)
+from flood_adapt.object_model.interface.scenarios import IScenario
+from flood_adapt.object_model.io.unitfulvalue import UnitfulLength, UnitTypesLength
+from flood_adapt.object_model.site import Site
+
+
+class TranslationModel(BaseModel):
+    """BaseModel describing the expected variables and data types for translation parameters of hurricane model."""
+
+    eastwest_translation: UnitfulLength = UnitfulLength(
+        value=0.0, units=UnitTypesLength.meters
+    )
+    northsouth_translation: UnitfulLength = UnitfulLength(
+        value=0.0, units=UnitTypesLength.meters
+    )
+
+
+class HurricaneEventModel(IEventModel):
+    """BaseModel describing the expected variables and data types for parameters of HistoricalHurricane that extend the parent class Event."""
+
+    ALLOWED_FORCINGS: dict[ForcingType, List[ForcingSource]] = {
+        ForcingType.RAINFALL: [
+            ForcingSource.CONSTANT,
+            ForcingSource.MODEL,
+            ForcingSource.TRACK,
+        ],
+        ForcingType.WIND: [ForcingSource.TRACK],
+        ForcingType.WATERLEVEL: [ForcingSource.MODEL],
+        ForcingType.DISCHARGE: [ForcingSource.CONSTANT, ForcingSource.SYNTHETIC],
+    }
+
+    hurricane_translation: TranslationModel
+    track_name: str
+
+
+class HurricaneEvent(IEvent):
+    MODEL_TYPE = HurricaneEventModel
+
+    attrs: HurricaneEventModel
+
+    def process(self, scenario: IScenario):
+        """Synthetic events do not require any processing."""
```

Comment:
??

Reply:
A hurricane event also requires an offshore model. But again, this workflow should be defined in the hazard class where we can also include SLR. The event class does not know anything about SLR.